Review images documentation. (#487)
MicroEJ/docs: ApplicationDeveloperGuide/UI/MicroUI/images.rst

The images heap is used to allocate the pixel data of:

- Mutable images (i.e. `BufferedImage`_  instances).
- Immutable images decoded at runtime, typically a PNG: the heap is used to store the decoded image **and** the runtime decoder's temporary buffers, required during the decoding step.
  After the decoding step, all the temporary buffers are freed.
  Note that the size of the temporary buffers depend on the decoder **and** on the original image itself (compression level, pixel encoding, etc.).
- Immutable images which are not byte-addressable, such as images opened with an input stream (i.e. `ResourceImage`_  instances).
- Immutable images which are byte-addressable but converted to a different output format (i.e. `ResourceImage`_  instances).

> In other words, every image which can not be retrieved using `ej.microui.display.Image.getImage()`_  is saved on the images heap.

Reviewer: ```suggestion
In other words, every image which cannot be retrieved using `ej.microui.display.Image.getImage()`_  is saved on the image heap.
```
